var/spack/repos/builtin/packages/mfem/package.py

```diff
@@ -163,8 +163,8 @@ class Mfem(Package):
     conflicts('timer=mpi', when='~mpi')
 
     depends_on('mpi', when='+mpi')
-    depends_on('hypre', when='+mpi')
-    depends_on('hypre@2.10.0:2.13.999', when='@:3.3.2')
+    depends_on('hypre@2.10.0:2.13.999', when='@:3.3.2+mpi')
+    depends_on('hypre@2.14:', when='@3.3.3:+mpi')
```

Comment:
Is it true that `mfem@3.3.3` can't be built with `hypre@2.13`? If concretization is the only problem, the first line of your change should be sufficient.

Reply:
3.3.3 is the internal number we use for development after 3.3.2, which currently is the same as `mfem@develop`. There will be no official release with that number, the next release will be 3.4.

(If you are curious, versions with even patch number correspond to official release, while odd patch numbers correspond to development)

So:
- `mfem@3.3.2` can be built with `hypre@2.13` but not `hypre@2.14`
- `mfem@develop` and `mfem@3.4` (when it comes out soon) can be built with any hypre version, including `hypre@2.14`